resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -340,6 +344,40 @@ private[spark] class Client(
     amContainer.setTokens(ByteBuffer.wrap(serializedCreds))
   }
 
+  /**
+   * Set configurations sent from AM to RM for renewing delegation tokens.
+   */
+  private def setTokenConf(amContainer: ContainerLaunchContext): Unit = {
+    // SPARK-37205: this regex is used to grep a list of configurations and send them to YARN RM
+    // for fetching delegation tokens. See YARN-5910 for more details.
+    // The feature is only supported in Hadoop 3.x and up, hence the check below.
+    val regex = sparkConf.get(config.AM_SEND_TOKEN_CONF)
+    if (regex.nonEmpty && VersionUtils.isHadoop3) {
+      logInfo(s"Processing token conf (spark.yarn.am.sendTokenConf) with regex $regex")
+      val dob = new DataOutputBuffer();
+      val copy = new Configuration(false);
+      copy.clear();
+      hadoopConf.asScala.foreach { entry =>
+        if (entry.getKey.matches(regex.get)) {
+          copy.set(entry.getKey, entry.getValue)
+          logInfo(s"Captured key: ${entry.getKey} -> value: ${entry.getValue}")
+        }
+      }
+      copy.write(dob);
+
+      // since this method was added in Hadoop 2.9 and 3.0, we use reflection here to avoid
```

Comment:
Are we doing this only for 3.x ? If not, relax the `isHadoop3` condition ?

Reply:
Yes, since Spark only supports built-in Hadoop 2.7 or 3.3, we have the check here. Do you mean support custom Hadoop version 2.9+ too with `-Phadoop.version=2.9.x`?

Reply:
Exactly - both 2.9 and 2.10 for example.

Reply:
Got it. I added the change.

Reply:
Gently ping @mridulm . Does the latest change look good to you?